File: 0049-group-anagrams/0049-group-anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
  hash = Hash.new {|hash, key| hash[key] = []}

  strs.each_with_index do |str, index|
    hash[str.chars.sort].push(str)
  end

  hash.values
end

File: 0049-group-anagrams/0049-group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> dic = new Dictionary<string, List<string>>();

        foreach(string str in strs){
          char[] chrArray = str.ToCharArray();
          Array.Sort(chrArray);
          string orderStr = new string(chrArray);

          if (dic.TryGetValue(orderStr, out List<string> value)){
            value.Add(str);
          }
          else{
            dic.Add(orderStr, new List<string>{str});
          }
        }


        return new List<IList<string>>(dic.Values);
    }
}